src/prefect/engine/task_runner.py

```diff
@@ -901,28 +901,28 @@ def get_task_run_state(
         except signals.LOOP as exc:
             new_state = exc.state
             assert isinstance(new_state, Looped)
-            new_state.result = Result(
-                value=new_state.result, result_handler=self.result_handler
+            new_state.result = Result( # todo: here
```

Comment:
Sounded from the call today that these need to be instantiated as `JSONResult` objects here as an internal implementation detail for looping.

Reply:
This data payload actually represents the output of the task, which we don't want to store as `JSON`.  We need to use the users `self.task.result` for this piece of data.